app/controllers/publishers/vacancies/build_controller.rb

```diff
@@ -23,10 +23,14 @@ def show
   def update
     if form.valid?
       update_vacancy
-      if session[:current_step] == :review
-        update_listing
+      if all_steps_valid? || save_and_finish_later?
+        if vacancy.published?
+          redirect_to organisation_job_path(vacancy.id), success: t("publishers.vacancies.show.success")
```

Comment:
redirect_to organisation_job_path(vacancy.id), success: (t("publishers.vacancies.show.success") if vacancy.published? )

Reply:
also check if you can put this logic into `redirect_to_next_step` method